build/source/engine/computFlux.f90

```diff
@@ -793,6 +793,9 @@ subroutine computFlux(&
                    err,cmessage)                   ! intent(out): error control
    if(err/=0)then; message=trim(message)//trim(cmessage); return; endif
 
+   ! compute total runoff (overwrite previously calculated value before considering aquifer)
+   scalarTotalRunoff  = scalarSurfaceRunoff + scalarAquiferBaseflow
+
```

Comment:
Is this double counting soil drainage? Note that `scalarSoilDrainage` is an input to the `bigAquifer` and is included in the total runoff calculation earlier

Reply:
I don't think so. This should just overwrite the earlier assignment.  In one case I tested, scalarRO decreased because the aquifer was delaying runoff, versus a case in which scalar RO included soil drainage but was being immediately expelled from the aquifer (aquiferBFrate = 0.1).

Reply:
But it is adding to the earlier assignment.... it doesn't look right to me.

Reply:
I think it is overwriting the earlier assignment, not adding to it.

Reply:
I really don't see this. `scalarRunoff` is calculated in soil hydrology (adding drainage) and then calculated again here (adding to it). I am probably missing something really obvious. This still does not seem right to me.

Reply:
The second statement doesn't add it to, but replaces its value, so that when you leave computFlux you end up with the value including the aquifer.  Stepping through it in gdb looks like this:  
(gdb) s
797        scalarTotalRunoff  = scalarSurfaceRunoff + scalarAquiferBaseflow
(gdb) p scalartotalrunoff 
$11 = 1.2939818142135093e-08
(gdb) p scalarSurfaceRunoff
$12 = 0
(gdb) p scalarAquiferBaseflow
$13 = 0.00022222222222222215
(gdb) s
814      if(ixCasNrg/=integerMissing) fluxVec(ixCasNrg) = scalarCanairNetNrgFlux/canopyDepth
(gdb) p scalarTotalRunoff
$14 = 0.00022222222222222215
(gdb) 
-----
I think if it were:  scalarTotalRunoff  = scalarTotalRunoff + scalarSurfaceRunoff + scalarAquiferBaseflow
then it would be adding to the earlier value.

Reply:
The second statement doesn't add it to, but replaces its value, so that when you leave computFlux you end up with the value including the aquifer.  Stepping through it in gdb looks like this:  
(gdb) s
797        scalarTotalRunoff  = scalarSurfaceRunoff + scalarAquiferBaseflow
(gdb) p scalartotalrunoff 
$11 = 1.2939818142135093e-08
(gdb) p scalarSurfaceRunoff
$12 = 0
(gdb) p scalarAquiferBaseflow
$13 = 0.00022222222222222215
(gdb) s
814      if(ixCasNrg/=integerMissing) fluxVec(ixCasNrg) = scalarCanairNetNrgFlux/canopyDepth
(gdb) p scalarTotalRunoff
$14 = 0.00022222222222222215
(gdb) 
-----
I think if it were:  scalarTotalRunoff  = scalarTotalRunoff + scalarSurfaceRunoff + scalarAquiferBaseflow
then it would be adding to the earlier value.

Reply:
Sorry, I don't know why it's going all bold here

Reply:
I could also be overlooking something as you certainly do more coding than I do.

Reply:
Yes, I was missing something obvious. I read scalarSurfaceRunoff as scalarTotalRunoff.